nestkernel/source_table.cpp

```diff
@@ -320,67 +364,59 @@ nest::SourceTable::get_next_target_data( const thread tid,
     current_source.set_processed( true );
```

Comment:
I don't think we need this second `current_source` variable here, we have it from above (without const).

And shouldn't we mark it as processed first after we have processed it, not here where we are about to begin processing it?

Reply:
i) see above
ii) yes, it should be marked as processed right before we return it, changed